test/test_server.c

```diff
@@ -172,12 +212,40 @@ int main_server(void)
 				printf("recv lua err\n");
 			}
 		}
+		sscanf(buf, "%s %s", method, path);
```

Comment:
The `recv()` call above is too artificial, in the sense that we are expecting the entire message in the first call. We should be more pragmatic and use a dynamic buffer and process the http header only when we read `\r\n\r\n`, which is the end of the header message by the protocol standard.

Reply:
Also, improve the parsing please.